crates/redeem/src/lib.rs

```diff
@@ -68,26 +70,39 @@ pub mod pallet {
 
     #[pallet::event]
     #[pallet::generate_deposit(pub(super) fn deposit_event)]
-    #[pallet::metadata(T::AccountId = "AccountId", Wrapped<T> = "Wrapped", Collateral<T> = "Collateral")]
+    #[pallet::metadata(DefaultVaultId<T> = "VaultId", T::AccountId = "AccountId", Wrapped<T> = "Wrapped", Collateral<T> = "Collateral")]
     pub enum Event<T: Config> {
         RequestRedeem(
-            H256,          // redeem_id
-            T::AccountId,  // redeemer
-            Wrapped<T>,    // redeem_amount_wrapped
-            Wrapped<T>,    // fee_wrapped
-            Collateral<T>, // premium
-            T::AccountId,  // vault_id
-            BtcAddress,    // user btc_address
-            Wrapped<T>,    // transfer_fee_btc
+            H256,              // redeem_id
+            T::AccountId,      // redeemer
+            Wrapped<T>,        // redeem_amount_wrapped
+            Wrapped<T>,        // fee_wrapped
+            Collateral<T>,     // premium
+            DefaultVaultId<T>, // vault_id
+            BtcAddress,        // user btc_address
+            Wrapped<T>,        // transfer_fee_btc
         ),
         // [redeemer, amount_wrapped]
         LiquidationRedeem(T::AccountId, Wrapped<T>),
         // [redeem_id, redeemer, amount_wrapped, fee_wrapped, vault, transfer_fee_btc]
-        ExecuteRedeem(H256, T::AccountId, Wrapped<T>, Wrapped<T>, T::AccountId, Wrapped<T>),
+        ExecuteRedeem(
+            H256,
+            T::AccountId,
+            Wrapped<T>,
+            Wrapped<T>,
+            DefaultVaultId<T>,
```

Comment:
We should probably also re-arrange this event by grouping the redeemer `AccountId` and the `VaultId` together.